keystone-init/templates/keystone-init-job.yaml

```diff
@@ -0,0 +1,45 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ template "fullname" . }}-job
+  labels:
+    app: {{ template "fullname" . }}
+    component: "{{ .Values.keystone_init.name }}"
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+spec:
+  template:
+    metadata:
+      labels:
+        app: {{ template "fullname" . }}
+        component: "{{ .Values.keystone_init.name }}"
+    spec:
+      restartPolicy: Never
+      volumes:
+        - name: preload-config
+          configMap:
+            name: "{{ template "fullname" . }}-preload"
+            items:
+              - key: preload.yml
+                path: preload.yml
+      containers:
+        - name: {{ template "fullname" . }}-job
+          image: "{{ .Values.keystone_init.image.repository }}:{{ .Values.keystone_init.image.tag }}"
+          imagePullPolicy: {{ .Values.keystone_init.image.pullPolicy }}
+          resources:
+{{ toYaml .Values.keystone_init.resources | indent 12 }}
+          env:
+            - name: LOG_LEVEL
+              value: {{ .Values.keystone_init.log_level }}
+            - name: KEYSTONE_TIMEOUT
+              value: "{{ .Values.keystone_init.timeout }}"
+            - name: KEYSTONE_VERIFY
+              value: "{{ .Values.keystone_init.verify }}"
+            - name: KEYSTONE_CERT
+              value: "{{ .Values.keystone_init.cert }}"
+{{ include "keystone_env" .Values.keystone_init.auth | indent 12 }}
```

Comment:
...and this is how the keystone_env template is included. just one line for all keystone/secret handling logic! :smile: